src/Store/Features.re

```diff
@@ -775,18 +823,54 @@ let update =
 
     switch (outmsg) {
     | Nothing => (state, Effect.none)
+    | BufferModifiedSet(id, _) =>
+      open Feature_Editor;
 
-    | CreateEditor({buffer, split, position, grabFocus}) =>
-      let editorBuffer = buffer |> Feature_Editor.EditorBuffer.ofBuffer;
-      let fileType = buffer |> Buffer.getFileType |> Buffer.FileType.toString;
+      let bufferId = id;
 
-      let editor =
-        Feature_Editor.Editor.create(
-          ~config=config(~fileType),
-          ~buffer=editorBuffer,
-          (),
+      let newLayout =
+        Feature_Layout.map(
+          editor =>
+            if (Editor.getBufferId(editor) == bufferId
+                && Editor.getPreview(editor) == true) {
+              Editor.setPreview(~preview=false, editor);
+            } else {
+              editor;
+            },
+          state.layout,
         );
 
+      ({...state, layout: newLayout}, Effect.none);
+
+    | CreateEditor({buffer, split, position, grabFocus, preview}) =>
+      open Feature_Editor;
+
+      let editorBuffer = buffer |> EditorBuffer.ofBuffer;
+      let fileType = buffer |> Buffer.getFileType |> Buffer.FileType.toString;
+
+      let bufferId = EditorBuffer.id(editorBuffer);
+
+      let existingEditor =
+        Feature_Layout.activeGroupEditors(state.layout)
+        |> List.find_opt(editor => Editor.getBufferId(editor) == bufferId);
+
+      let (isPreview, editor) =
+        switch (existingEditor) {
+        | Some(ed) =>
+          let isPreview = Editor.getPreview(ed) && preview;
+
+          (isPreview, Editor.setPreview(~preview=isPreview, ed));
+        | None => (
+            preview,
+            Editor.create(
+              ~config=config(~fileType),
+              ~buffer=editorBuffer,
+              ~preview,
+              (),
+            ),
+          )
+        };
+
```

Comment:
It seems like this is breaking the split functionality:

![broken-split](https://user-images.githubusercontent.com/13532591/99008216-93801200-24fa-11eb-9ddf-a30f5f9f634d.gif)

In particular - note that when I split using `:vsp` - it seems like a common editor is being used. When I scroll, both editors scroll, and likewise when I move the screen. So this looks like a regression.

Reply:
Got you, will check it later.

Reply:
Looks like this is fixed now in the latest - thanks @fanantoxa !